packages/react-date-time/src/components/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,253 @@
+import * as React from 'react';
+import flatpickr from 'flatpickr';
+import { css } from '@patternfly/react-styles';
+import FlatpickrLanguages from 'flatpickr/dist/l10n';
+import 'flatpickr/dist/themes/light.css';
+import '@patternfly/patternfly/patternfly-date-picker.css';
+import styles from '@patternfly/react-styles/css/components/DatePicker/date-picker';
+import { DatePickerLocales } from './DatePickerUtils';
+import { TextInput } from '@patternfly/react-core/dist/js/components/TextInput/TextInput';
+
+export interface DatePickerProps
+  extends Omit<React.HTMLProps<HTMLInputElement>, 'onChange' | 'onFocus' | 'onBlur' | 'disabled' | 'ref'> {
+  /** Additional classes added to the date time picker. */
+  className?: string;
+  /** Flag indicating whether the value of the date picker is invalid. */
+  invalid?: boolean;
+  /** Accessible label for the date picker */
+  'aria-label'?: string;
+  /** The minimum date that a user can start picking from. */
+  minDate?: string;
```

Comment:
I think we should use the same types as flatpickr
https://github.com/flatpickr/flatpickr/blob/f4478d27fce40532298a738aab428b3bbbff751c/src/types/options.ts#L165 for all the props that are forwarded

Reply:
So, flatpickr works best when you require the user to choose a date from the calendar. There are a few requirements in our design - especially the one that wants the user to be able to type in their date rather than be forced to use the calendar, that begin to pose some problems.

flatpickr uses weird date formatting tokens - not standard ones, so we simplify things quite a bit.
we do our own error handling for invalid date formats since flatpickr does not do very robust error handling. 

The last example on this page: https://flatpickr.js.org/examples/ sheds the very beginning of some light.

since we are allowing the user to type in their own input, which could be invalid - we handle the basic error handling, which includes parsing and formatting, before handing dates off to flatpickr

Reply:
yes, you can listen to input changes, and call this.calendar.parseDate(inputValue, formatToCheckAgainst) simply, no?

Reply:
I guess the trade off is instead of writing our own parser/formatter functions, we would need to create a mapping for the datepickr formatter tokens to standard tokens. Or link to the flatpickr doumentation in our docs, but I'd rather keep flatpickr more under the hood.

Reply:
https://flatpickr.js.org/formatting/